### ISSUE TITLE
Eligible_clients_audit table does not show initial values for client status.

### ISSUE THREAD

**Issue by CTAJavier** [hserv/coordinated-entry#284]
We are trying to use the Eligible_clients_audit table to determine when clients were added/removed, from the active list/ eligible_clients table. However, we seem to have a few clients that are listed as having Ignore_match_process Set to TRUE as their only record. This should be impossible.

A client is added to the eligible clients table, once there is a score posted for the client as a result of completing the VI SPDAT, at that time the client is eligible for matching, so Ignore_match_process is set to False by the System, so there should be no way for a client to have ignore_match_process as TRUE as their only record.

This means that we have no record of the original values or original dates that a client was added. That information should be in the table for us to be able to audit the eligible_client table.

**Comment by eric-jahn** [comment]
@logicsandeep is checking on this now

**Comment by logicsandeep** [comment]
@CTAJavier  Can you please provide me a client_id for this issue please ?

**Comment by CTAJavier** [comment]
examples sent through slack
*Javier Celedon*
*Operations Director*
Direct: (408) 437-9168
Main: (408) 437-8800
*Facebook <http://www.facebook.com/ctagroup.org> • Twitter
<https://twitter.com/CTA_Group_> • Video
<http://www.ctagroup.org/what-we-do/video/>*

<http://www.ctagroup.org/>

**Comment by logicsandeep** [comment]
@suryayadavalli , 
Javier has provided us the cliendids in the home app channel and It is very clear that  we only have 1 record in in the eligible_clients_audit table. Can you please take a look at the trigger you created ?

Assigning this issue to you also.

**Comment by suryabi** [comment]
@jceledon: looked into the issue above.  Essentially,  Audit keeps track of the snapshot of the record that's changed. You will always have the latest record in *eligible_clients* table and *eligible_clients_audit* will have all the older versions of the *eligible_client*.

The reason you do not have the same record in *eligible_clients* and *eligible_clients_audit* is that - the later has the historic / older record.

**Comment by suryayadavalli** [comment]
**Root cause of the issue.**
There was an issue that we fixed earlier due to *bonus_score* was set to NULL  (_the default value was supposed to be 0_)  and Monterrey score processing job was failing due to that.  So, before this issue was fixed, there were eligible clients with bonus scores set to NULL, which are not considered by the Audit logging (_with an assumption that we always have some value in that column_).  So, audit logging for those clients whose bonus score was initially NULL did not happen.  This issue shouldn't have occurred, after we made the fix on April,15, 2019.

I made changes now to capture the initial record , so audit log always provides every thing, including the latest. They do not have to query two entities (audit log and audit log history) for complete picture for the client.
